jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/EventsHandler.java

```diff
@@ -0,0 +1,419 @@
+            if (chunk == null)
+                c = null;
+            else if (chunk.hasRemaining())
+                c = Content.Chunk.asChunk(chunk.getByteBuffer().asReadOnlyBuffer(), chunk.isLast(), chunk);
+            else
+                c = chunk.isLast() ? Content.Chunk.EOF : Content.Chunk.EMPTY;
+            onRequestRead(wrapped, c);
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onRequestRead", x);
+        }
+    }
+
+    private void fireOnAfterHandling(Request request, boolean handled, Throwable failure)
+    {
+        try
+        {
+            onAfterHandling(request, handled, failure);
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onAfterHandling", x);
+        }
+    }
+
+    private void fireOnResponseBegin(Request request, HttpFields response, int status)
+    {
+        try
+        {
+            onResponseBegin(request, status, response);
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onResponseBegin", x);
+        }
+    }
+
+    private void fireOnResponseWrite(Request request, boolean last, ByteBuffer content)
+    {
+        try
+        {
+            onResponseWrite(request, last, content.asReadOnlyBuffer());
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onResponseWrite", x);
+        }
+    }
+
+    private void fireOnResponseWriteComplete(Request request, Throwable failure)
+    {
+        try
+        {
+            onResponseWriteComplete(request, failure);
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onResponseWriteComplete", x);
+        }
+    }
+
+    private void fireOnResponseTrailersComplete(Request request, HttpFields trailers)
+    {
+        try
+        {
+            onResponseTrailersComplete(request, trailers);
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onResponseTrailersComplete", x);
+        }
+    }
+
+    private void fireOnComplete(Request request, Throwable failure)
+    {
+        try
+        {
+            onComplete(request, failure);
+        }
+        catch (Throwable x)
+        {
+            LOG.info("Error firing onComplete", x);
+        }
+    }
+
+    /**
+     * Invoked just before calling the server handler tree (i.e. just before the {@link Runnable}
+     * returned from {@link org.eclipse.jetty.server.HttpChannel#onRequest(MetaData.Request)} is run).
+     *
+     * <p>
+     * This is the final state of the request before the handlers are called.
+     * This includes any request customization.
+     * </p>
+     *
+     * @param request the request object. The {@code read()}, {@code demand(Runnable)} and {@code fail(Throwable)} methods must not be called by the listener.
+     * @see org.eclipse.jetty.server.HttpChannel#onRequest(MetaData.Request)
+     */
+    protected void onBeforeHandling(Request request)
+    {
+    }
```

Comment:
Should these all have LOG.debugs in them?  This would make this handler nice to have for debugging even on its own.

Reply:
Added debug logs.